ios/brave-ios/Sources/BraveWallet/Crypto/Stores/TransactionDetailsStore.swift

```diff
@@ -100,6 +100,15 @@ class TransactionDetailsStore: ObservableObject, WalletObserverStore {
     txServiceObserver = nil
   }
 
+  func updateTransaction(_ transaction: BraveWallet.TransactionInfo) {
+    guard transaction.id == self.transaction.id else {
+      // not the transaction currently open
+      return
+    }
+    self.transaction = transaction
+    self.update()
```

Comment:
why not just change the status instead of a whole assignment and then calling update()? nothing really changed except this tx's status?

Reply:
We use `ParsedTransaction` for display. To avoid duplicating most of `update()` we could implement a cache for Solana estimated tx fee & price fetch for the transaction and avoid second network call, unless we'd prefer fresh data?